subsys/bluetooth/mesh/access.c

```diff
@@ -999,7 +999,7 @@ static int bt_mesh_vnd_mod_msg_cid_check(const struct bt_mesh_model *mod)
 	for (op = mod->op; op->func; op++) {
 		cid = (uint16_t)(op->opcode & 0xffff);
 
-		if (cid == mod->vnd.company) {
+		if (BT_MESH_MODEL_OP_LEN(op->opcode) == 3 && cid == mod->vnd.company) {
```

Comment:
I don't think this is correct since the force option is to exactly ensure that a vendor model doesn't serve messages opcode with different Company ID.

Since the spec clearly allows a vendor model serve 1-/2-octets opcodes, this should only check matching Company ID for 3-octets opcode messages. Therefore, I think the check should be like this:
```
if (BT_MESH_MODEL_OP_LEN(op->opcode) == 3 && cid != mod->vnd.company) {
    LOG_ERR("...");
    return -EINVAL;
}
```